Applications/Classification/jni/main_func.cpp

```diff
@@ -0,0 +1,305 @@
+ * @bug		No known bugs except for NYI items
+ * @brief	This is Classification Example with one FC Layer
+ *              The base model for feature extractor is mobilenet v2 with
+ * 1280*7*7 feature size. It read the Classification.ini in res directory and
+ * run according to the configureation.
+ *
+ */
+#include <climits>
+#include <cmath>
+#include <fstream>
+#include <iomanip>
+#include <iostream>
+#include <queue>
+#include <sstream>
+#include <stdlib.h>
+#include <time.h>
+
+#include "layers.h"
+#include "neuralnet.h"
+#include "tensor.h"
+
+#define TRAINING true
+
+/**
+ * @brief     Data size for each category
+ */
+const unsigned int total_train_data_size = 100;
+
+const unsigned int total_val_data_size = 10;
+
+const unsigned int total_test_data_size = 100;
+
+const unsigned int buffer_size = 100;
+
+/**
+ * @brief     Number of category : Three
+ */
+const unsigned int total_label_size = 10;
+
+/**
+ * @brief     Max Epoch
+ */
+const unsigned int iteration = 3000;
+
+const unsigned int mini_batch = 32;
+
+const unsigned int feature_size = 62720;
+
+using namespace std;
+
+/**
+ * @brief     location of resources ( ../../res/ )
+ */
+string data_path;
+
+bool duplicate[total_label_size * total_train_data_size];
+bool valduplicate[total_label_size * total_val_data_size];
+
+/**
+ * @brief     step function
+ * @param[in] x value to be distinguished
+ * @retval 0.0 or 1.0
+ */
+float stepFunction(float x) {
+  if (x > 0.9) {
+    return 1.0;
+  }
+
+  if (x < 0.1) {
+    return 0.0;
+  }
+
+  return x;
+}
+
+/**
+ * @brief     Generate Random integer value between min to max
+ * @param[in] min : minimum value
+ * @param[in] max : maximum value
+ * @retval    min < random value < max
+ */
+static int rangeRandom(int min, int max) {
+  int n = max - min + 1;
+  int remainder = RAND_MAX % n;
+  int x;
+  do {
+    x = rand();
+  } while (x >= RAND_MAX - remainder);
+  return min + x % n;
+}
+
+/**
+ * @brief     load data at specific position of file
+ * @param[in] F  ifstream (input file)
+ * @param[out] outVec
+ * @param[out] outLabel
+ * @param[in] id th data to get
+ * @retval true/false false : end of data
+ */
+bool getData(std::ifstream &F, std::vector<float> &outVec,
+             std::vector<float> &outLabel, int id) {
+  uint64_t position = (feature_size + total_label_size) * id * sizeof(float);
+  if ((position + (feature_size + total_label_size) * sizeof(float)) >
+      ULLONG_MAX)
+    return false;
+  F.seekg(position, std::ios::beg);
+  for (unsigned int i = 0; i < feature_size; i++)
+    F.read((char *)&outVec[i], sizeof(float));
+  for (unsigned int i = 0; i < total_label_size; i++)
+    F.read((char *)&outLabel[i], sizeof(float));
+
+  return true;
+}
+
+/**
+ * @brief      get data which size is mini batch for train
+ * @param[out] outVec
+ * @param[out] outLabel
+ * @param[out] status for error handling
+ * @retval true/false
+ */
+bool getMiniBatch_train(std::vector<std::vector<std::vector<float>>> &outVec,
```

Comment:
It might be better off to allow this function to have `std::string filename` as an argument.

Reply:
I don't think so. It is the use case to use function to get the data set. It does not care about how the user provide the data. It could be other way like calculate directly from image or get data from other sources, not from raw file. If someone want to use raw file, then he can specify file name in configuration file.